pkg/prometheus/promcfg.go

```diff
@@ -628,7 +629,7 @@ func (cg *configGenerator) generateProbeConfig(
 	ignoreNamespaceSelectors bool,
 	enforcedNamespaceLabel string) yaml.MapSlice {
 
-	jobName := fmt.Sprintf("%s/%s", m.Namespace, m.Name)
+	jobName := fmt.Sprintf("%s/%s/%d", m.Namespace, m.Name, i)
 	if m.Spec.JobName != "" {
 		jobName = m.Spec.JobName
```

Comment:
this is the actual problem, we need to not set the job name directly, we need to instead relabel the `job` label in relabel configs